islandora.module

```diff
@@ -611,6 +612,10 @@ function islandora_permission() {
       'title' => t('Revert datastream history'),
       'description' => t('Revert to a previous version of a datastream.'),
     ),
+    ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS => array(
```

Comment:
I like your first suggestion. Give them a warning. There are situations this is an expected behavior.